src/Rules/AnyOfRule.php

```diff
@@ -40,12 +40,12 @@ public function processValue($value, Args $args, FieldContext $context)
 				);
 				$anyValidationSucceeded = true;
 			} catch (ValueDoesNotMatch | InvalidData $exception) {
-				$type->overwriteInvalidSubtype($key, $exception->getInvalidType());
+				$type->overwriteInvalidSubtype($key, $exception);
 			}
 		}
 
 		if (!$anyValidationSucceeded) {
-			throw ValueDoesNotMatch::create($type);
+			throw ValueDoesNotMatch::create($type, $value);
```

Comment:
Same tests as for AllOfRule is welcome. Mutation of value shouldn't matter in this case but let's be future self proof

Reply:
TODO: test case

Reply:
Test case added